src/main/java/net/snowflake/ingest/streaming/internal/SnowflakeStreamingIngestChannelInternal.java

```diff
@@ -112,13 +117,21 @@ class SnowflakeStreamingIngestChannelInternal implements SnowflakeStreamingInges
                     String.format("%s_%s", name, channelSequencer),
                     0,
                     this.owningClient.getAllocator().getLimit());
-    this.arrowBuffer = new ArrowRowBuffer(this);
+    this.rowBuffer = createRowBuffer(client);
     this.encryptionKey = encryptionKey;
     this.encryptionKeyId = encryptionKeyId;
     this.onErrorOption = onErrorOption;
     logger.logInfo("Channel={} created for table={}", this.channelName, this.tableName);
   }
 
+  private RowBuffer<T> createRowBuffer(SnowflakeStreamingIngestClientInternal<T> client) {
+    // TODO: The circular dependency SnowflakeStreamingIngestChannelInternal <-> RowBuffer
+    // can be probably reconsidered
+    //noinspection unchecked
```

Comment:
do you have a better idea? let's just remove the comment or do it now, I feel like this is something that we will never address if not doing it now :)

Reply:
I agree :) I did not what to further explode this PR, it does not look to me as a-couple-of-lines change.
I created a JIRA for this SNOW-657667 and try to get to this afterwards.